Added Operations team charter. (#5)
tim-schilling/dsf-working-groups: active/operations.md

# Operations Team

## Scope of responsibilities

The Operations (Ops) Team is responsible for managing sensitive matters such as credentials. The dedicated team helps limit attack surface area and limit impact of incidents.

The Ops team manages several platforms:
- The server running [djangoproject.com](https://djangoproject.com) and [Django's Trac instance](https://code.djangoproject.com/)

> - The Jenkins server for CI

Reviewer: ```suggestion
- The [Jenkins server for CI](https://djangoci.com/)
```

The Ops team also helps manage several applications due to their privileged access:

- The [Django GitHub organization](https://github.com/django)
- The [DjangoCon GitHub organization](https://github.com/djangocon)
- DNS for djangoproject.com and other domains
- Fastly CDN

> - Django's 1Password organization

Reviewer: Perhaps we should add the DSF Stripe account here? or is it implied from the access to 1Password?

Author: My definition of "manage" is that you're the group that is emailed when something needs to change or be reviewed. If that's what the team is doing for Stripe, then I agree it should be added here.

I don't think we need to include everything you have access to.

Reviewer: Agreed in the Ops call that the team does not truly manage Stripe.

Author: Django Snippets runs on Heroku.

The Ops team is able to make the following decisions based on their judgement:

> - Platform updates

Reviewer 2: ```suggestion
- Software and OS upgrades (major and minor)
- Updates to or migrations between established service providers
```

- Performance improvements
- Bug fixes

> - Backwards compatible changes

Reviewer 2: ```suggestion
- Backwards compatible changes
- Access control requests
- DNS changes
```

The Ops team will request Board approval before moving forward any other decision.
## Initial membership

- [Optional] Board Liaison (must be an active Board member; may be the same as Chair/Co-Chair):
- Members:
    - Baptiste Mispelon
    - Jacob Walls
    - Mariusz Felisiak
    - Markus Holtermann
    - Natalia Bidart
    - Sarah Boyce
    - Tobias McNulty

## Future membership

> The DSF Board selects new members by voting. If the team determines it needs new members, a call for volunteers will be posted on the [djangoproject.com blog](https://www.djangoproject.com/weblog/) and the [Django Forum](https://forum.djangoproject.com).

Author: ```suggestion
The team manages its own membership by invitation. If the team needs to make a call for volunteers, it will be posted on the [djangoproject.com blog](https://www.djangoproject.com/weblog/) and the [Django Forum](https://forum.djangoproject.com).
```

The DSF Board selects new members by voting. If the team determines it needs new members, a call for volunteers will be posted on the [djangoproject.com blog](https://www.djangoproject.com/weblog/) and the [Django Forum](https://forum.djangoproject.com).

The membership will operate as follows:
- All Django Fellows are automatically added to the Ops team

> - A Django Fellow contract termination removes the person from the Ops team

Reviewer: This is unclear to me, Mariusz remained in the team after retiring.

Author: I copied this from the mergers/releasers team. I figure we should align on this. I didn't want to include the extra language of "temporarily reviewed until the Board can review their access" because it's implied that the board could say vote that this person is on the team and avoid the removal.

It's a matter of trying to keep the language simple and allowing people the flexibility to get things done.

Reviewer: During the Ops calls, it was clarified that Mariusz was part of the team before becoming a Fellow, so my comment is not relevant at this point.

Reviewer: (tried to resolve but couldn't)

- Each year, every member will need to reaffirm their membership with the team

### Qualifications

The primary qualification for Ops team members is trust. The person must be trusted with the highest level of access in the community. This means they need to have a steady, extensive track record of good decision-making and dedication to the community.

From a technical perspective members are expected to have knowledge in the following areas:

- DevOps experience

>   - Ansible

Reviewer 2: ```suggestion
- DevOps experience (Ansible, AWS, OpenStack)
```

>   - DNS

Reviewer 2: ```suggestion
- DNS management
```

  - Ansible
  - DNS
- Server admin

> 

Reviewer: Is the goal in this section to list all expected knowledge? If yes we could add:
- docker
- postgresql

Author: Would those both fall under DevOps? Or just Docker under DevOps with postgres on its own?

Reviewer: Can be dropped since is too detailed.

Reviewer 2: I'm okay with including them--it feels a little short as-is. 😅 Suggestion added. I think no nesting might be simpler to follow.

> - Server admin

Reviewer 2: ```suggestion
- Linux server administration
- Docker/Podman
- PostgreSQL
```

## Budget

> The Ops team manages 100k worth of infrastructure through donations from in-kind sponsors.

Reviewer 2: ```suggestion
The Ops team manages USD $100,000-plus worth of infrastructure through in-kind donations from DSF sponsors.
```

The Ops team manages 100k worth of infrastructure through donations from in-kind sponsors.

There is no dedicated budget at this time, but funds can be requested from the Board.

## Comms

The team communications and can be contacted via the [ops@djangoproject.com](mailto:ops@djangoproject.com) email.

The team can be mentioned in the following ways:

- [GitHub team](https://github.com/orgs/django/teams/ops-team)

> The team meets monthly via Meet.

Reviewer: The team also has a private channel in Discord, and I think it could be a good idea to maybe move to Slack to have a consistent story for the various WGs.

Author: Resolving as there was agreement to stick with how things are and not adopt Slack.

# Operations Team

## Scope of responsibilities

> The Operations (Ops) Team is responsible for managing sensitive matters such as credentials. The dedicated team helps limit attack surface area and limit impact of incidents.

Reviewer 2: ```suggestion
The Operations (Ops) Team is responsible for managing the DSF's core infrastructure, hosting and software-as-a-service providers, and sensitive matters such as credentials. The dedicated team helps limit attack surface area and limit impact of incidents.
```

The Ops team manages several platforms:
- The server running [djangoproject.com](https://djangoproject.com) and [Django's Trac instance](https://code.djangoproject.com/)
- The Jenkins server for CI

The Ops team also helps manage several applications due to their privileged access:

- The [Django GitHub organization](https://github.com/django)
- The [DjangoCon GitHub organization](https://github.com/djangocon)
- DNS for djangoproject.com and other domains
- Fastly CDN

> - Django's 1Password organization

Reviewer: Move all these to an onboarding doc? Yes! Use instead:
```suggestion
- Hosting providers
- Content Delivery Network
- Secret management
```

Reviewer 2: I'm okay keeping DNS in the list.

The Ops team is able to make the following decisions based on their judgement:

> - Minor platform upgrades and updates

Reviewer: ```suggestion
- Platform updates
```

> - Minor platform upgrades and updates

Reviewer 2: ```suggestion
- Software and OS upgrades (major and minor)
- Updates to or migrations between established service providers
```

- Performance improvements
- Bug fixes
- Backwards compatible changes

> The Ops team will request Board approval before moving forward any other decision.

Reviewer 2: What about sharing secrets with and making DNS changes on behalf of existing community members? We regularly receive such requests and evaluate them based on our own judgment without explicit approval from the Board. That being said, I'm also not sure we want to advertise publicly that the community can email the Ops team for a DNS change or access to a secret.

Author: > What about sharing secrets with and making DNS changes on behalf of existing community members?

That's a good catch. I think we should have a line here that captures the service-based nature of the team. What about adding "Handling access control requests" to the list above?

> That being said, I'm also not sure we want to advertise publicly that the community can email the Ops team for a DNS change or access to a secret.

I suspect the potential harm here is minimal. The opening line about managing secrets is what makes the group a target. I think pointing out that this is the group of people the rest of the community relies on for making changes would be common knowledge or a very reasonable assumption.

Reviewer 2: Sounds good, thanks! Added to my pending edit.

> 

Reviewer: ```suggestion

```

## Initial membership

- [Optional] Board Liaison (must be an active Board member; may be the same as Chair/Co-Chair):

> - Members members:

Reviewer 3: ```suggestion
- Members:
```

Author: Or we could have gone with Members²

    - Baptiste Mispelon
    - Jacob Walls
    - Mariusz Felisiak
    - Markus Holtermann
    - Natalia Bidart
    - Sarah Boyce
    - Tobias McNulty

## Future membership

> The DSF Board selects new members by voting. If the team determines it needs new members, a call for volunteers will be posted on the [djangoproject.com blog](https://www.djangoproject.com/weblog/) and the [Django Forum](https://forum.djangoproject.com).

Author: Change this to invite only by the team to match existing process.

The membership will operate as follows:

> - All Django Fellows are automatically added to the Ops team

Reviewer: ```suggestion
- Django Fellows are included in the Ops team mailing list for awareness, and are invited to be part of the Ops team
```

- A Django Fellow contract termination removes the person from the Ops team
- There should be at least three members at all times
- There is no upper limit to the number of members

> - Each year, every non-Fellow member will need to reaffirm their membership with the team

Reviewer: ```suggestion
- Each year, every member will need to reaffirm their membership with the team
```

### Qualifications

The primary qualification for Ops team members is trust. The person must be trusted with the highest level of access in the community. This means they need to have a steady, extensive track record of good decision-making and dedication to the community.

> From a technical perspective members are expected to have knowledge in the following areas:

Reviewer 3: Have or build knowledge? I've never used ansible, but I think I'm still in a position to learn the deployment tooling we use.

Reviewer: This would be more important if the Fellows would be automatically added to the team. Can be resolved as is.
